go-backend/db/organization.go

```diff
@@ -0,0 +1,216 @@
+package db
+
+import (
+	"context"
+	"time"
+	"regexp"
+	logger "github.com/sirupsen/logrus"
+)
+
+const (
+	CreateOrganizationQuery = `INSERT INTO organizations (
+		name,
+		contact_email,
+		domain_name,
+		subscription_status,
+		subscription_valid_upto,
+		hi5_limit,
+		hi5_quota_renewal_frequency,
+		timezone,
+		created_by,
+		updated_by)
+		VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10)`
+
+	UpdateOrganizationQuery = `UPDATE organizations SET (
+		name,
+		contact_email,
+		domain_name,
+		subscription_status,
+		subscription_valid_upto,
+		hi5_limit,
+		hi5_quota_renewal_frequency,
+		timezone,
+		updated_by,
+		updated_on) = 
+		($1, $2, $3, $4, $5, $6, $7, $8, $9, $10) where id = $11`
+
+	DeleteOrganizationQuery = `DELETE FROM organizations WHERE id = $1`
+	GetOrganizationQuery = `SELECT id,
+		name,
+		contact_email,
+		domain_name,
+		subscription_status,
+		subscription_valid_upto,
+		hi5_limit,
+		hi5_quota_renewal_frequency,
+		timezone,
+		created_by,
+		created_on,
+		updated_by,
+		updated_on FROM organizations WHERE id=$1`
+	
+	ListOrganizationsQuery =`SELECT id,
+		name,
+		contact_email,
+		domain_name,
+		subscription_status,
+		subscription_valid_upto,
+		hi5_limit,
+		hi5_quota_renewal_frequency,
+		timezone,
+		created_by,
+		created_on,
+		updated_by,
+		updated_on FROM organizations ORDER BY name ASC`
+	emailRegex = 	"^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$"
+	domainRegex = `(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]`
+)
+
+type Organization struct {
+	ID                       int       `db:"id" json:"id" `
+	Name                     string    `db:"name" json:"name"`
+	ContactEmail             string    `db:"contact_email" json:"email"`
+	DomainName               string    `db:"domain_name" json:"domain_name"`
+	SubscriptionStatus       int       `db:"subscription_status" json:"subscription_status"`
+	SubscriptionValidUpto    time.Time `db:"subscription_valid_upto" json:"subscription_valid_upto"`
+	Hi5Limit                 int    `db:"hi5_limit" json:"hi5_limit"`
+	Hi5QuotaRenewalFrequency string    `db:"hi5_quota_renewal_frequency" json:"hi5_quota_renewal_frequency"`
+	Timezone                 string    `db:"timezone" json:"timezone"`
+	CreatedBy                int    `db:"created_by" json:"_"`
+	CreatedOn                time.Time `db:"created_on" json:"_"`
+	UpdatedBy                int    `db:"updated_by" json:"_"`
+	UpdatedOn                time.Time `db:"updated_on" json:"_"`
+}
+
+//TODO how to declare this as reusable
+type ErrorResponse struct {
+	Code 	string `json:"code"`
+	Message	string `json:"message"`
+	Fields map[string]string `json:"fields"`
+}
+
+func (org *Organization) ValidateOrganization() (errorResponse map[string]ErrorResponse, valid bool) {
+
+	valid = true
```

Comment:
its not a good practise to start with assuming that record is valid. Instead work on a reverse model where you only set valid = true when you are sure that there are no errors, in this case at the end of function.

So at the end of func you can check:
```
if len(fieldErrors) == 0 {
  valid = true
  return
}```